src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -97,7 +97,17 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 var result = await binder.BindModelAsync(bindingContext);
                 if (result != null)
                 {
-                    return result;
+                    // Use returned ModelBindingResult if it either indicates model was set or is related to errors.
+                    // The second condition is necessary because some binders add errors that would be duplicated when
```

Comment:
I'll generalize this comment because `TypeConverterModelBinder` and `TypeMatchModelBinder` both include calls to `bindingContext.ModelState.SetModelValue()`.  in `TypeConverterModelBinder`'s case, that may be associated with returning a result with `!IsModelSet`.

in any case this code is about whether to maintain the current result or `null`.  it will exit the loop regardless.

the difference between returning `null` and a result with `!IsModelSet` (the only case where the added `ContainsKey()` check matters) is that the caller may fall back to the empty prefix in the first case but will not in the second.

further extra entries in the `ModelState` will lead to `!IsValid` since validation won't change their `Unvalidated` state.